iceoryx_posh/include/iceoryx_posh/internal/popo/used_chunk_list.hpp

```diff
@@ -21,161 +22,107 @@
+            , offset(MAX_OFFSET)
         {
-            // does the entry match the one we want to remove?
-            if (m_data[current] != nullptr && m_data[current]->m_chunkHeader == f_chunkHeader)
-            {
-                // return the chunk mgmt entry as SharedChunk object
-                f_chunk = mepoo::SharedChunk(m_data[current]);
-                m_data[current] = nullptr;
-
-                // remove index from used list
-                if (current == m_usedListHead)
-                {
-                    m_usedListHead = m_list[current];
-                }
-                else
-                {
-                    m_list[previous] = m_list[current];
-                }
-
-                // insert index to free list
-                m_list[current] = m_freeListHead;
-                m_freeListHead = current;
-
-                //@todo can we do this cheaper with a global fence in cleanup?
-                m_synchronizer.clear(std::memory_order_release);
-                return true;
-            }
-            previous = current;
+            static_assert(sizeof(relativ_ptr_data) <= 8U, "The relativ_ptr_data size must not exceed 64 bit!");
+            static_assert(std::is_trivially_copyable<relativ_ptr_data>::value,
+                          "The relativ_ptr_data must be trivially copyable!");
         }
-        return false;
-    }
-
-    // only once from runtime context
-    void setup()
-    {
-        m_synchronizer.test_and_set(std::memory_order_acquire);
-        /// @todo maybe also call init here?
-    }
-
-    // from RouDi context once the applications walked the plank
-    void cleanup()
-    {
-        m_synchronizer.test_and_set(std::memory_order_acquire);
-
-        for (auto& data : m_data)
+        constexpr relativ_ptr_data(uint16_t segment, uint64_t offset) noexcept
+            : segment(segment)
+            , offset(offset & MAX_OFFSET)
         {
-            if (data != nullptr)
-            {
-                mepoo::SharedChunk{data};
-            }
+            cxx::Ensures(offset < MAX_OFFSET && "offset must not exceed MAX:OFFSET!");
         }
 
-        init(); // just to save us from the future self
-    }
+        uint16_t segment : 16;
+        uint64_t offset : 48;
 
-  private:
-    void init()
-    {
-        // build list
-        for (uint32_t i = 0u; i < Size; ++i)
+        void reset() noexcept
         {
-            m_list[i] = i + 1u;
+            constexpr relativ_ptr_data NULLPTR_EQUIVALENT{};
+            *this = NULLPTR_EQUIVALENT;
         }
 
-        if (Size > 0)
-        {
-            m_list[Size - 1u] = InvalidIndex; // just to save us from the future self
-        }
-        else
+        bool isNullptr() const noexcept
         {
-            m_list[0u] = InvalidIndex;
+            return segment == MAX_SEGMENT && offset == MAX_OFFSET;
         }
 
+        static constexpr uint64_t MAX_SEGMENT{std::numeric_limits<uint16_t>::max()};
```

Comment:
Those constants probably belong somewhere else where the shared memory parameters/bounds are defined. But ok for the time being I guess.